woocommerce/woocommerce-android
# Issue/3725 Clean up Stripe sdk on logout and on site change

Parent issue #3725

The PR implements cleaning payments-related data when a user switches sites or logs out.

From [the documentation](https://stripe.dev/stripe-terminal-android/com/stripe/stripeterminal/Terminal.html#clearCachedCredentials--)


```
In order to switch accounts in your app:
* if a reader is connected, call disconnectReader
* call clearCachedCredentials
* call discoverReaders and connectReader to connect to a reader. The connectReader call will request a new connection token from your backend server.
```

### How To Test
Settings -> Manage readers -> Connect -> Logout -> Login -> Connect again
and
Settings -> Manage readers -> Connect -> Switch sites -> Connect

Combinations like Connect -> Disconnect Switch sites also nice to check.


https://user-images.githubusercontent.com/4923871/120643465-deee1380-c47e-11eb-9f6f-f61e961d6469.mp4
